tests/test_robot_logger.cpp

```diff
@@ -54,21 +60,84 @@ class TestRobotLogger : public ::testing::Test
                                             first_action_timeout,
                                             max_number_of_actions);
         frontend = std::make_shared<Frontend>(data);
-        logger = std::make_shared<Logger>(data);
 
         auto tmp_dir = std::filesystem::temp_directory_path();
-        logfile = tmp_dir / "test_robot_logger.dat";
+        logfile = tmp_dir / "test_robot_logger.log";
     }
 
     void TearDown() override
     {
         // clean up
         std::filesystem::remove(logfile);
     }
+
+    /**
+     * @brief Check whether the given CSV log is valid.
+     *
+     * @param filename  The log file.
+     * @param start_t  Expected first time index in the log.
+     * @param end_t  Expected last time index in the log.
+     * @param use_gzip  If true, assume the file is gzip-compressed.
+     */
+    void check_csv_log(const std::string &filename,
+                       uint32_t start_t,
+                       uint32_t end_t,
+                       bool use_gzip)
+    {
+        std::string line;
+
+        // open file, potentially with gzip decompression
+        std::ifstream infile_raw(filename);
+        boost::iostreams::filtering_istream infile;
+        if (use_gzip)
+        {
+            infile.push(boost::iostreams::gzip_decompressor());
+        }
+        infile.push(infile_raw);
+
+        // check header
+        {
+            ASSERT_TRUE(std::getline(infile, line))
+                << "Failed to read header line";
+
+            // remove trailing whitespaces
+            line.erase(line.find_last_not_of(" \n") + 1);
+
+            ASSERT_EQ(
+                line,
+                "#time_index timestamp status_action_repetitions "
+                "status_error_status observation_values_0 observation_values_1 "
+                "applied_action_values_0 applied_action_values_1 "
+                "desired_action_values_0 desired_action_values_1");
+        }
+
+        // check data
+        for (uint32_t i = start_t; i < end_t; i++)
+        {
+            int time_index;
+            double time_stamp, action_repetitions, error_status, obs_0, obs_1,
+                applied_0, applied_1, desired_0, desired_1;
+
+            ASSERT_TRUE(std::getline(infile, line))
+                << "Failed to read line " << i;
```

Comment:
I am bit confused: in which stream this is writing ?

Reply:
This is the error message that is displayed if the assert fails.